docs/how-to/hip_runtime_api/memory_management/host_memory.rst

```diff
@@ -0,0 +1,205 @@
+  :keywords: AMD, ROCm, HIP, host memory
+
+.. _host_memory:
+
+********************************************************************************
+Host memory
+********************************************************************************
+
+Host memory where on the host (e.g. CPU) of the machine in random access memory
+(RAM). The host memory has three different main types in HIP:
+
+* Pageable memory
+* Pinned memory
+* Unified Memory
+
+These different types of memory should be used at different use cases. The
+pageable and pinned memory using explicit memory management, where the
+developers have direct control over memory operations, while at the unified
+memory case the developer gets a simplified memory model with less control over
+low level memory operations.
+
+The data transfer differences between the pageable or pinned memory usage
+represented in the next figure.
+
+.. figure:: ../../../data/how-to/hip_runtime_api/memory_management/pageable_pinned.svg
+
+The unified memory management and explicit memory management main difference
+highlighted at the following figure.
+
+.. figure:: ../../../data/how-to/hip_runtime_api/memory_management/unified_memory/um.svg
+
+The unified memory management described at the :doc:`/how-to/hip_runtime_api/memory_management/unified_memory`.
+
+Pageable memory
+================================================================================
+
+Pageable memory is exists on "pages" (blocks of memory), which can be
+migrated to other memory storage. For example, migrating memory between CPU
+sockets on a motherboard, or a system that runs out of space in RAM and starts
+dumping pages of RAM into the swap partition of your hard drive.
+
+Pageable memory is usually gotten when calling ``malloc`` or ``new`` in a C++
+application. The following example shows the pageable host memory usage in HIP.
+
+.. code-block:: cpp
+
+  #define ELEMENT_NUMBER 100
+
+  int *host_input, *host_output;
+  // Host allocation
+  host_input  = new int[ELEMENT_NUMBER];
+  host_output = new int[ELEMENT_NUMBER];
+
+  // Host data preparation
+  for (int i = 0; i < ELEMENT_NUMBER; i++) {
+    host_input[i] = i;
+  }
+  memset(host_output, 0, ELEMENT_NUMBER * sizeof(int));
+
+  int *device_input, *device_output;
+
+  // Device allocation
+  hipMalloc((int **)&device_input, ELEMENT_NUMBER * sizeof(int));
+  hipMalloc((int **)&device_output, ELEMENT_NUMBER * sizeof(int));
+
+  // Device data preparation
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyHostToDevice);
+  hipMemset()
+
+  // Run the kernel
+  ...
+
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyDeviceToHost);
+
+  // Free host memory
+  delete[] host_input;
+  delete[] host_output;
+
+  // Free device memory
+  hipFree()
+  hipFree()
+
+.. note::
+
+  :cpp:func:`hipMalloc` and :cpp:func:`hipFree` are blocking calls, however, HIP
+  recently added non-blocking versions :cpp:func:`hipMallocAsync` and
```

Comment:
Recently? Better not to mention time in documentation, rather "also provides" (or if you know at what rocm version they have been added, feel free to mention that here)